### ISSUE TITLE
[v3] Implement new Concept Exercise: enumeration

### ISSUE THREAD

**Issue by verdammelt** [exercism/common-lisp#312]
This issue describes how to implement the `enumeration` concept exercise for the Common Lisp track.

## Getting started

**Please please please read the docs before starting.** Posting PRs without reading these docs will be a lot more frustrating for you during the review cycle, and exhaust Exercism's maintainers' time. So, before diving into the implementation, please read up on the following documents:

- [The features of v3](https://github.com/exercism/v3/blob/master/docs/concept-exercises.md).
- [Rationale for v3](https://github.com/exercism/v3/blob/master/docs/rationale-for-v3.md).
- [What are concept exercise and how they are structured?](https://github.com/exercism/v3/blob/master/docs/features-of-v3.md)

Please also watch the following video:

- [The Anatomy of a Concept Exercise](https://www.youtube.com/watch?v=gkbBqd7hPrA).

## Goal

The goal of this exercise is to teach the student about how to do [enumeration] in Common Lisp. To keep things simple this exercise will use only require a few of `loop`'s features (although they could also be done with members of the `do` "family").

## Learning objectives

- Know the existence of `loop`.
- Know how to use "for-as-arithmetic" clause
- Know how to use "for-as-in-list" clause
- Know how to use "for-as-across" clause
- Know how to use `collect` "list-accumulation" clause
- Know how to use `summing` "numeric-accumulation" clause
- Know how to use `if` and `else` "conditional" clause.

## Out of scope

The following will be out of scope for this exercise - but should be mentioned in the `after.md`.

- Other parts of `loop`.
- The `do` "family" of functions.

## Concepts

- `enumeration`
- `loop.basic`

## Prerequisites

Student will not be taught these topics but will need some familiarity with them to solve the problems. The introduction, instructions and stub code could be written to lower the need of these prerequisites.

- arithmetic
- strings
- ??? more depending upon the story of the exercise.

## Resources to refer to

- [Loop for Black Belts][loop-for-black-belts]
- [Loop Facility][hyperspec-loop]

For example:

### Hints

Hints for the user can include specific examples of what they need to do with `loop`.

### After

The `after.md` file should mention/discuss the following:

- Other parts of `loop`. No need to be exhaustive but perhaps should include `into`, `finally`, `return`, other "for-as" clauses
- Mention/overview of `do` "family" of functions.

## Representer

The representer will need to be able to normalize the `loop` macro.

## Analyzer

The analyzer will need to be able to recognize `loop` vs. `do` and give the user some guidance to use `loop` for this exercise. This guidance should not stop the solution from being accepted.

## Implementing

The intention of focusing on `loop` for this exercise is to keep things simple and reduce prerequisites. `loop` hides things like assignment needed for incrementing and collection than `do` does.

Compare:

    `(loop for i from 0 upto 10 collect i)` 

vs.

    `(do ((i 0 (1+ i)) (result (list) (cons i result))) ((> i 10) (nreverse result)))`
    
### Suggestion for exercise tasks

- The student must sum even integers in a range. (for-as-arithmetic)
- The student must square all numbers in a list. (for-as-in-list)
- The student must find every vowel in a string. (for-as-across)

## Help

If you have any questions while implementing the exercise, please post the questions as comments in this issue.

[enumeration]: ./reference/concepts/enumeration.md
[hyperspec-loop]: http://www.lispworks.com/documentation/HyperSpec/Body/06_a.htm
[loop-for-black-belts]: http://www.gigamonkeys.com/book/loop-for-black-belts.html

**Comment by verdammelt** [comment]
I removed the enumeration concept in favor of iteration. Maybe we'll add this back later - but for now closing this issue.
